src/components/EndChatButton.tsx

```diff
@@ -0,0 +1,22 @@
+import * as React from 'react';
+import * as FlexWebChat from "@twilio/flex-webchat-ui";
+import { ButtonBase } from '@material-ui/core';
+
+type Props = {
+  channelSid: string;
+}
+
+export default function EndChatButton ({channelSid}: Props) {
+  
+  const handleEndChat = () => {
+    FlexWebChat.Actions.invokeAction("RestartEngagement");
+    FlexWebChat.Actions.invokeAction("MinimizeChat");
+    // FlexWebChat.Actions.invokeAction("SendMessage", { body: "User ended the conversation", channelSid: channelSid });
+  }
+
```

Comment:
I tried to invoke ending / exiting the conversation in UI AND send a message to Flex. Namely using promises, separate statements in anonymous functions. However, this has not worked.

Currently. I can do only one of the two. Looking for suggestions on how to do both in a clean way.

Reply:
Have you been trying to do this in the order shown here? I'd suspect that you can't invoke a "SendMessage" action after removing the user from the chat channel. It's possible also that use of `await` is necessary to make sure an action finishes before another, otherwise race conditions could appear. You could look at the Developer Console to see if any errors are popping up during this.